signup page

aakankshaagr opened the pull request:
### Description

<!--Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change.-->
Developed SignUp Page

Fixes #248 

### Type of Change:
<!--**Delete irrelevant options.**-->

- Code
- Quality Assurance
- User Interface


**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)



### How Has This Been Tested?
<!--Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.-->
![image](https://user-images.githubusercontent.com/71868375/123910526-30af8e00-d998-11eb-81cd-95314d8da99e.png)



### Checklist:
<!--**Delete irrelevant options.**-->

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials


**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

aakankshaagr committed:
signup page

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -136,71 +171,92 @@ export default function Register() {
+                    <label id="password">Password :</label>
+                    <input
+                      aria-labelledby="password"
+                      className="feilds"
```
please fix typo

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -136,71 +171,92 @@ export default function Register() {
+                      className="feilds"
+                      type={isPasswordShown ? "text" : "password"}
+                      name="password"
+                      placeholder="atleast 8 characters"
```
```suggestion
                      placeholder="at least 8 characters"
```

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -136,71 +171,92 @@ export default function Register() {
+                    <label id="confirmPassword">Confirm Password :</label>
+                    <input
+                      aria-labelledby="confirmPassword"
+                      className="feilds"
```
typo

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -136,71 +171,92 @@ export default function Register() {
+                      className="feilds"
+                      type={isPasswordShown ? "text" : "password"}
+                      name="confirmPassword"
+                      placeholder="atleast 8 characters"
```
although the design shows this as placehloder, IMO it makes more sense if we show `re-type password` here instead since this field is not to insert a totally new password.

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -2,126 +2,161 @@ import React, { useState, useContext } from "react";
+    <div className="container">
+      <div class="row align-items-start">
+        <div class="col">
+          <img src="src/assets/images/side_image.jpg" alt="background" />
```
instead of using image path and have the image saved inside images folder, you can use the githubusercontent link of an image as it was used on the [About.jsx line 12](https://github.com/anitab-org/bridge-in-tech-web/blob/develop/src/home/About.jsx).

mtreacy002 reviewed (changes requested):
@aakankshaagr , thank you for submitting the PR. I have left some feedback for you below as well as an alternative to upload the side image. Also, please don't forget to remove all the commented lines that are no longer needed. Let me know if you have any questions 😉.

aakankshaagr committed:
Responsive Signup with Footer

aakankshaagr commented:
Thanks @mtreacy002 !! I have made changes as you suggested. About image I don't know how to get side_image in way suggested by you. Thankfully, somehow it worked and image is displayed. I have made changes in classes naming as there was lot of css getting included with those names.
Icon part is left as m facing some issues in icon display.
I didn't find any information for page layout when screen size is small so I made it as per my wish.If it's already there then please inform me, I will make changes according to that.
In media queries, I want screen to be divided in 2 rows top having image and bottom having form. I don't know why image is not showing up.Can you please point out where I am going wrong?

Waiting for your feedback :) !!

aakankshaagr commented:
![image](https://user-images.githubusercontent.com/71868375/123908703-85053e80-d995-11eb-8002-8adda29515aa.png)
if Register component can too stretch like footer and navbar then it will look better.
![image](https://user-images.githubusercontent.com/71868375/123910263-d1ea1480-d997-11eb-8c33-67fa080dbd58.png)

naveen8801 committed:
UI: Added Video Section on Homepage (#261)

* Video Section Added

* Minor changes in video section

mtreacy002 commented:
@aakankshaagr , is there any reason you don't make it stretch like the footer? I believe the image in Figma also shows that the container (image+form) is stretchable. 
<img width="595" alt="Screen Shot 2021-07-08 at 11 34 09 pm" src="https://user-images.githubusercontent.com/29667122/124930832-023f4c00-e045-11eb-98b2-a518a2397099.png">

mtreacy002 reviewed (changes requested):
Hi @aakankshaagr , the build compiled with warnings. Can you try to fix this, please? Thanks 
<img width="595" alt="Screen Shot 2021-07-08 at 11 22 59 pm" src="https://user-images.githubusercontent.com/29667122/124931336-75e15900-e045-11eb-81b2-d8b7d94b4139.png">

aakankshaagr committed:
added preloader

aakankshaagr committed:
made resgiter dynamic

aakankshaagr commented:
@mtreacy002 On stretching image got doubled and that wasn't looking good. So, as I mentioned earlier ,I felt like only making form stretchable.

aakankshaagr committed:
added password icon

aakankshaagr committed:
Merge branch 'develop' into signup

aakankshaagr committed:
removed unwanted changes of route.js

aakankshaagr committed:
added password icon

aakankshaagr committed:
formatted register.jsx

aakankshaagr commented:
@mtreacy002 route.js got autoformatted and by mistake, I pushed route.js too. While trying to fix that I somehow merged develop.I don't know how to undo this. I tried to rebase and git reset --merge but it is not working. Please suggest some other way to remove this.

mtreacy002 reviewed (changes requested):
@aakankshaagr , I checked your code changes and it doesn't seem to have major issues. Auto-formatting of routes.jsx seems ok and the merged of develop branch to your pr branch seems to have no impact to the code you're submitting. Although, I do have some feedback for you to work on:
* the build is failing. Below is the screenshot
<img width="1394" alt="Screen Shot 2021-07-16 at 1 59 39 pm" src="https://user-images.githubusercontent.com/29667122/125890808-6e2d7b55-cd25-4a5c-9b51-1d6e96fa01bb.png">

* the tests are also failing. see this [gist](https://gist.github.com/mtreacy002/9d8b61db394ff97e7c142be36f4eaa6b) for the error warnings.

Can you please fix these errors? Thanks

PS: I have a second thought on the form stretch. The stretching of the register form field seems odd since the fields on the form won't need that bigger input field. I think the form stretch on the bigger screen size should be maximized to up to 75% of the screen width instead to make it reasonable. Perhaps your screenshot [above](https://github.com/anitab-org/bridge-in-tech-web/pull/262#issuecomment-871119935) was made with smaller zoom range (~50 - 67%) which is probably not what most of our users would have (~75-125%). So, you can put it back to the way it was IMO 😉.
<img width="1657" alt="Screen Shot 2021-07-16 at 2 17 25 pm" src="https://user-images.githubusercontent.com/29667122/125891580-c194451b-95c1-402c-a18d-06a9e69e9d22.png">

aakankshaagr committed:
removed errors in register.jsx

aakankshaagr committed:
removed warnings from register

aakankshaagr commented:
@mtreacy002  I have removed error warnings. It is also suggesting we should add alt text for the password icon image. What shall I keep as alt text?
Form size is dynamic, as screen size increases fields will stretch. So I think changes help in case if the user screen size is way large or else it will display like previously I showed you.

vj-codes commented:
@mtreacy002 hey can you please approve the workfow run in this PR?

aakankshaagr commented:
Tests are failing bcz placeholder texts are not matching in tests and register files. So I am changing placeholder texts. @mtreacy002 According to your earlier suggestion password placeholder values isn't suitable for adding in tests. So I am changing them to be the same as label names. Is that fine or I should do vice versa?

aakankshaagr committed:
removed build errors

aakankshaagr committed:
removed test error

aakankshaagr commented:
@mtreacy002 All test cases are passed :) 
![image](https://user-images.githubusercontent.com/71868375/130120090-d8620b5c-8d81-42d7-a92c-070a072cef80.png)
I hope now it passes all build tests.

aakankshaagr commented:
@mtreacy002  can you please approve the workflow?

aakankshaagr committed:
added alt text in images

aakankshaagr committed:
user login succesfully test case passed

aakankshaagr commented:
![image](https://user-images.githubusercontent.com/71868375/131217956-5152c580-457c-4eab-bd9b-c63955663047.png)

![image](https://user-images.githubusercontent.com/71868375/131218069-0778a46e-d786-4318-9c40-ff87527c9e11.png)

All test cases are passing along with no warnings. @mtreacy002 Can you plz approve the workflow.

aakankshaagr commented:
@mtreacy002 can you plz approve the workflow?

aakankshaagr committed:
changed layout for <40% zoom

aakankshaagr commented:
@mtreacy002  please approve the workflow. I have changed the layout when zoom<50% as suggested by you. All test cases have been passed locally.

mtreacy002 commented on src/register/Register.jsx:
```diff
@@ -1,291 +1,301 @@
 export default function Register() {
-    const {isAuth} = useContext(AuthContext);
-    const [isValidName, setIsValidName] = useState(true);
-    const [isValidUsername, setIsValidUsername] = useState(true);
```
We need to keep this field on the form since the Mentorship System has this field in its signup form.
<img width="845" alt="Screen Shot 2021-11-24 at 9 44 47 am" src="https://user-images.githubusercontent.com/29667122/143141370-fa0aa725-4f6a-4b1b-9ca2-0066b357e435.png">

So in your SignUp form design, please add the username field to match the form in Mentorship System (as you can see on the existing code on BIT develop branch)

<img width="1263" alt="Screen Shot 2021-11-24 at 9 44 28 am" src="https://user-images.githubusercontent.com/29667122/143141545-fb94500f-e874-4cc1-b49e-2ab72936a463.png">
<img width="1262" alt="Screen Shot 2021-11-24 at 8 59 19 am" src="https://user-images.githubusercontent.com/29667122/143142267-cf4182fc-a689-4eb1-927a-cce278a86c26.png">

PS 1: Not having the same fields causing error (invalid payload) when user tried to register as can be seen on the screenshot above.
PS 2: Please fix the location of error response. It should be on a separate line (not on the same line as "Available as" text). You can place it on top of the SignUp button.

Also, for consistency across MS-BIT applications, you can you please follow the existing order for the fields? So start with Name, then Username, followed by Email, then continue with the rest.

mtreacy002 reviewed (changes requested):
Hi @aakankshaagr , thank you for the pr and my apology for replying late. 
Please check the below feedback.
Plus, can you please keep the Sidebar and the VideoSection from the existing code in your commits. It seems like you've deleted them by accident on this pr.

Thank you beforehand 😉

aakankshaagr committed:
added username in registration and tests

aakankshaagr committed:
wrapped react usestate in act in test file

aakankshaagr committed:
added act() in testcases

aakankshaagr committed:
changed position of payload error

aakankshaagr commented:
> Hi @aakankshaagr , thank you for the pr and my apology for replying late. Please check the below feedback. Plus, can you please keep the Sidebar and the video section from the existing code in your commits. It seems like you've deleted them by accident on this pr.
> 
> Thank you beforehand 😉

It's fine @mtreacy002 😊. Sure, I will add them again. I am sorry I thought that the VideoSection part was not related to my pr and I added them by mistake. So I removed them. Thanks to other contributors for correcting my mistake.I have added username field and updated tests cases too. Also repositioned error before signup button. Can you please check the changes once again?
Also, I found that footer is not working. Link tag is missing `to` attribute and this is causing a warning in the register form section. Please look into it once and tell me if there is something I can do about it.

aakankshaagr commented:
![image](https://user-images.githubusercontent.com/71868375/156898450-0a459f6a-a8de-4a18-b6ed-5044e81e31ba.png)
@mtreacy002 is this matching with your suggested changes? Please review this and let me know if there are some changes that need to be done.
Thank you

rkraeher commented on src/register/Register.jsx:
```diff
@@ -1,291 +1,331 @@
-    const [showTermsModal,setShowTermsModal] = useState(false);
+  const { isAuth } = useContext(AuthContext);
+  const [isValidUsername, setIsValidUsername] = useState(true);
+  const [isValidName, setIsValidName] = useState(true);
```
if these are validators, does it make more sense for initial state to be set to `false` instead? 🤔  maybe they should only be `true` after successful validation?

aakankshaagr replied:
@rkraeher Thank you for reviewing. Values are set so when the user enters the wrong value then display an error. Apart from validating, error message is also handled by this.

rkraeher commented on src/register/Register.jsx:
```diff
@@ -1,291 +1,331 @@
+              <section className="register">
+                <h1>Welcome To Bridge In Tech</h1>
+                <h4>
+                  Lets start your account setup so you can have access to a
```
Lets is a typo. should be Let's

rkraeher commented on src/tests/PersonalDetails.test.js:
```diff
@@ -1,144 +1,212 @@
-    expect(screen.getByLabelText("Photo Url :", {selector: "input"})).toBeEmpty();
-    expect(screen.getByLabelText("Resume Url :", {selector: "input"})).toBeEmpty();
+    expect(
+      screen.getByLabelText("Username :", { selector: "input" })
```
there is a lot of weird formatting and line breaks in this file that really decreases the readability of the tests.

rkraeher commented on src/tests/Register.test.js:
```diff
@@ -1,202 +1,321 @@
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+import {
+  render,
```
something odd happened with the formatting in this file as well. this negatively affects readability like ☝️

aakankshaagr replied:
I guess it's because prettier is not working properly for me. I guess the code will be readable after running the prettier. I tried hard to correct this even done manually but after saving everything goes back to the same.So it's quite difficult to format tests.

rkraeher replied:
I see. do you have prettier extension or any other linter enabled? it is annoying to do it manually but if you do that and have the prettier disabled it shouldn't autoformat when you save

rkraeher commented on src/tests/Register.test.js:
```diff
@@ -1,202 +1,321 @@
+    target: { value: "My Name" },
+  });
+  act(() => {
+    fireEvent.change(screen.getByPlaceholderText("Username"), {
```
do we really need to use `act()` in this test if none of these form input events are asynchronous except for the submit registration?

aakankshaagr replied:
@rkraeher I guess we need to use act() 
![image](https://user-images.githubusercontent.com/71868375/156923790-7ce72586-bcf6-4d51-af48-3e33f8551ace.png)

rkraeher replied:
this warning seems to have to do with state updates happening asynchronously. this is not really how `act` should be used if you check out the [docs](https://reactjs.org/docs/test-utils.html#act). 

[This article](https://davidwcai.medium.com/react-testing-library-and-the-not-wrapped-in-act-errors-491a5629193b) also has a really useful explanation of the error and some suggestions on alternative solutions for how to deal with it effectively and eliminate the warning

rkraeher reviewed (commented):
nice work 🙂 just a refactor suggestion: there are a lot of repeated assertions throughout the tests where the only thing that changes are the values you are passing. you could remove a lot of duplicated code by mapping these repeated values with the jest `it.each` method. [see the docs](https://jestjs.io/docs/api#testeachtablename-fn-timeout)

aakankshaagr committed:
removed typo mistake
